jniwrapper/native/src/org_jitsi_modified_sctp4j_SctpJni.c

```diff
@@ -425,14 +437,41 @@ JNI_OnLoad(JavaVM *vm, void *reserved)
 
                 if (sendCb)
                 {
-                    clazz = (*env)->NewGlobalRef(env, clazz);
-                    if (clazz)
+                    jmethodID logDebugCb
+                        = (*env)->GetStaticMethodID(
+                               env,
+                               clazz,
+                               "logDebug",
+                               "(Ljava/lang/String;)V");
+
+                    jmethodID logInfoCb
+                        = (*env)->GetStaticMethodID(
+                               env,
+                               clazz,
+                               "logInfo",
+                               "(Ljava/lang/String;)V");
+
+                    jmethodID logErrorCb
+                        = (*env)->GetStaticMethodID(
+                               env,
+                               clazz,
+                               "logError",
+                               "(Ljava/lang/String;)V");
+
+                    if (logDebugCb && logInfoCb && logErrorCb)
                     {
-                        Sctp_clazz = clazz;
-                        Sctp_receiveCb = receiveCb;
-                        Sctp_sendCb = sendCb;
-                        Sctp_vm = vm;
-                        r = JNI_VERSION_1_4;
+                        clazz = (*env)->NewGlobalRef(env, clazz);
+                        if (clazz)
```

Comment:
Why the nested `if`s? Can't we just set the different callbacks independently (I think the rest of the code checks each one before they are used)? Should we print something if one of them is not available?

Reply:
I was just copying what the existing code was doing.